### DISCUSSION TITLE
Export Users and passwords

### DISCUSSION THREAD

**Discussion by mattWalters0** [supabase/supabase#3897] [Questions] [answered]
If we ever need to migrate away from Superbase will it be possible to export users with the hashed passwords so they could be reimported to another platform?  See link below for an example:

https://firebase.google.com/docs/cli/auth#authexport

**Comment by awalias** [comment, marked as answer]
yes! all the auth data (emails, hashed passwords etc.) are all stored in the `auth` schema in your database (you can see the hashed passwords using `select * from auth.users;`)

when you do a pg_dump using your postgres connection string you will also get the entire auth schema and data (check here for a known issue with permissions + workaround: https://github.com/supabase/supabase/issues/1856)

also as an FYI supabase auth just uses the standard bcrypt `GenerateFromPassword` when hashing passwords https://github.com/supabase/gotrue/blob/1dddc248fb820e638f0d54b9d51c2803295b5c1d/models/user.go#L245

**Comment by mattWalters0** [comment]
Thanks for confirming @awalias !
